resources/dw/catalog/resource_catalog.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2024 Cloudera. All Rights Reserved.
+//
+// This file is licensed under the Apache License Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/cdp"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+)
+
+type dwCatalogResource struct {
+	client *cdp.Client
+}
+
+var (
+	_ resource.Resource              = (*dwCatalogResource)(nil)
+	_ resource.ResourceWithConfigure = (*dwCatalogResource)(nil)
+)
+
+func NewDwCatalogResource() resource.Resource {
+	return &dwCatalogResource{}
+}
+
+func (r *dwCatalogResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.client = utils.GetCdpClientForResource(req, resp)
+}
+
+func (r *dwCatalogResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dw_catalog"
+}
+
+func (r *dwCatalogResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = dwDefaultCatalogSchema
+}
+
+func (r *dwCatalogResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var plan resourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	clusterID := plan.ClusterID.ValueStringPointer()
+
+	if !(plan.PollingOptions != nil && plan.PollingOptions.Async.ValueBool()) {
+		callFailedCount := 0
+		stateConf := &retry.StateChangeConf{
+			Pending:      []string{"Accepted", "Creating", "Created", "Loading", "Starting"},
+			Target:       []string{"Running"},
+			Delay:        30 * time.Second,
+			Timeout:      plan.getPollingTimeout(),
+			PollInterval: 30 * time.Second,
+			Refresh:      r.stateRefresh(ctx, clusterID, &callFailedCount, plan.getCallFailureThreshold()),
+		}
+		_, err := stateConf.WaitForStateContext(ctx)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error waiting for Data Warehouse catalog",
```

Comment:
Public doc uses the terminology as `Database Catalog` let's stick to that one.
https://docs.cloudera.com/data-warehouse/cloud/getting-started/topics/dw-get-started-data-lake-defaultdbc.html

Reply:
changed